### PR TITLE
Write auth unit tests

### PR DESCRIPTION
**Pull request by iFrostizz**
Write some unit tests for the currently untested auth package and move all of the auth variants into their own package.

### CODE REVIEW

File: auth/consts.go [hunk 1]
```diff
@@ -3,18 +3,10 @@
...
 	ED25519ID   uint8 = 0
 	SECP256R1ID uint8 = 1
 	BLSID       uint8 = 2
 )
-
-func Engines() map[uint8]vm.AuthEngine {
-	return map[uint8]vm.AuthEngine{
-		ED25519ID: &ED25519AuthEngine{},
-	}
-}
```
Comment (iFrostizz) [inline auth/consts.go:20]: This function was unused and now creates an import cycle. It was also probably not correctly placed.

File: auth/bls/bls_test.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/crypto"
+	"github.com/ava-labs/hypersdk/crypto/bls"
+)
+
+func TestBLSSign(t *testing.T) {
```
Comment (joshua-kim) [inline auth/bls/bls_test.go:16]: For tests like this it's idiomatic to write a table-driven test ([ref](https://go.dev/wiki/TableDrivenTests)) to make it easy to add/modify tests.

I suspect that we can get a decent table-driven test by using a table like:
```
type test struct {
    name string
    message []byte
    signature []byte
    expected error
}
```

If we do this we should be able to clean up this test - since this test technically covers two different cases (one where the signature is valid, and one where the signature is invalid) and we should be testing for one.

File: auth/ed25519/ed25519_test.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	for i, signedMessage := range signedMessages {
+		verifyFunc := verifier.Add(signedMessage.Msg, signedMessage.Rauth)
+		if (i+1)%batchSize == 0 {
+			require.NotNil(verifyFunc)
+			err := verifyFunc()
+			require.NoError(err)
+		} else {
+			require.Nil(verifyFunc)
+		}
+	}
```
Comment (iFrostizz) [inline auth/ed25519/ed25519_test.go:80]: It's not clear how `Done` should be called here. Since we need to know the batch size to know how much signature need to be aggregated, we already know when a batch is full and that the verification function pointer is returned.
This makes `Done` irrelevant.

File: auth/consts.go (hunk 1 above)
Comment (iFrostizz) [reply auth/consts.go:20]: Oh actually it's called `examples/morpheusvm/controller/controller.go`, let's find a better place

Comment (iFrostizz) [reply auth/consts.go:20]: Moved it to `auth/common`, let me know if it's a good choice.

File: auth/bls/bls_test.go (hunk 2 above)
Comment (iFrostizz) [reply auth/bls/bls_test.go:16]: Sure, will do! 
Maybe for this purpose we could write fuzz tests, because it seems hard to think about messages that would find edge cases since they should all verify as expected, also the signature cannot really be guessed so we can only cheat here by checking the output and writing it as the test input.

Comment (iFrostizz) [reply auth/bls/bls_test.go:16]: But absolutely fine for few cases that checks if verify works/fails as expected ! Thanks!

Comment (joshua-kim) [reply auth/bls/bls_test.go:16]: Hmm yeah you're right. In that case you can either have a `setup func() ([]byte, []byte)` that generates the message + signature pair, or we can just split the current UT into two UTs (valid/invalid). We can also do fuzz testing like you mentioned... i think these are all reasonable so pick whatever you think looks best.

Comment (iFrostizz) [reply auth/bls/bls_test.go:16]: So I have changes the bls test suite to have an `expectedSignature` in addition to the `signature`.
The `signature` will overwrite the signature that gets written by blst when calling `Sign` so that we can check that the verification fails as expected. Not sure it it's the best solution.

File: auth/secp256r1/secp256r1_test.go [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/crypto"
+	"github.com/ava-labs/hypersdk/crypto/secp256r1"
+)
+
+func TestSecp256r1Sign(t *testing.T) {
+	require := require.New(t)
+
+	priv, err := secp256r1.GeneratePrivateKey()
```
Comment (iFrostizz) [inline auth/secp256r1/secp256r1_test.go:19]: It seems pretty hard to generate a deterministic secp256r1 key.
